docs/explanations/packaging-data.md

```diff
@@ -0,0 +1,158 @@
+
+- Packaging shapes taxonomy : https://github.com/openfoodfacts/openfoodfacts-server/blob/main/taxonomies/packaging_shapes.txt
+- Packaging materials taxonomy : https://github.com/openfoodfacts/openfoodfacts-server/blob/main/taxonomies/packaging_materials.txt
+- Packaging recycling taxonomy : https://github.com/openfoodfacts/openfoodfacts-server/blob/main/taxonomies/packaging_recycling.txt
+- Preservation methods taxonomy (related) : https://github.com/openfoodfacts/openfoodfacts-server/blob/main/taxonomies/preservation.txt
+
+Those taxonomies are used to structure packaging data in Open Food Facts, and to analyze unstructured input data.
+
+## How we could improve it
+
+### Extend the attributes of the packaging components in the "packagings" data structure
+
+#### Weight
+
+We need to add an attribute for the weight of the packaging component. We might need to add different fields to distinguish values that have been entered by users that weight the packaging, versus values provided by the manufacturer, or average values that we have determined from other products, or that we got from external sources.
+
+### Make the "packagings" data structure READ and WRITE
+
+The "packagings" data structure is currently a READ only field. We could create an API to make it a READ and WRITE field.
+
+For new products, clients (website and apps) could ask users to enter data about all packaging components of the product.
+
+For existing products, clients could display the packaging components and let users change them (e.g. adding or removing components, entering values for new attributes, editing attributes to add more precise values (e.g. which type of plastic) etc.).
+
+#### Add a way to indicate that the "packagings" data structure contains all the packaging components of the product
+
+We currently have no way to know if the packaging data we have for a product is complete, or if we may be missing some packaging components.
+
+We could have a way (e.g. a checkbox) that users could use to indicate all components are accounted for. And we could also do the reverse, and indicate that it is very likely that we are missing some packaging components (e.g. if we have a "cap" but no other component to put the cap on).
+
+### Deprecate the "packaging" tags field
+
+We could discard the existing "packaging" tags field, and replace it with an API to allow clients to add partial information about packaging components.
+
+For instance, if Robotoff detects that the product is in plastic bottle by analyzing a product photo, it could send {shape:"bottle", material:"en:plastic"} and it would be added / combined with the existing "packagings" data.
+
+### Keep the "packaging_text_[language code]" field
+
+It is important to keep this field, as we can display it as-is, use it as input data, and it may contain interesting data that we do not analyze yet.
+
+When filled, the values for this field can be analyzed and added to / combined with the "packagings" data structure.
```

Comment:
Explain if there would be a round trip between this field and packaging_text_[language code].
That is if writing in packagings should affect packaging_text ?

Could we also keep a track of which characters (span) of packaging_text mapped to an entry in packagings ?

How do you deal with eventual parts of packaging_text that you are not able to parse ? Do you reject the input in this case ?

Reply:
Writing packagings will not affect packaging_text.

We can keep track of what part of the text was matched to what. What we don't recognize is ignored.

Added notes about that in the doc.